setup.py

```diff
@@ -1,8 +1,96 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
-from setuptools import setup, find_packages
+import os
+import sys
+from setuptools import setup, find_packages, Extension, Command
+from setuptools.command.build_ext import build_ext
 
+
+# ---------------------------------------------------------------------------- #
+#
+def join_path_list(path_list):
+    path = path_list[0]
+    path_list.remove(path)
+    for _path in path_list:
+        path = os.path.join(path, _path)
+    return path
+
+
+# ---------------------------------------------------------------------------- #
+#
+def TomoPyLibraryFile():
+    """
+    Determines the library output file
+    """
+
+    libname = None
+    if sys.platform.lower().startswith('win'):
+        libname = 'libtomopy.dll'
+    elif sys.platform == 'darwin':
+        libname = 'libtomopy.dylib'
+    else:
+        libname = 'libtomopy.so'
+
+    path = join_path_list([os.getcwd(), "tomopy", "sharedlibs", libname])
```

Comment:
Instead of using all these cases, you could use `glob` and a regular expression.
```python
glob.glob('./tomopy/sharedlibs/libtomopy.*')
```

Reply:
That could work for the second part. I think the first check would be recommended to be certain that you give priority to the platform specific library since this could cause problems with Docker -- e.g. `docker run -it --rm -v ${PWD}:/home/tomopy tomopy-dev ./pyctest_tomopy.py` -- would return `libtomopy.dylib` on my Mac if that still existed, which is entirely possible because there isn't an easy `make clean` available.

Reply:
Good point. I didn't consider that you would be building for multiple platforms on one machine.